pallene/coder.lua

```diff
@@ -1035,20 +1035,34 @@ gen_cmd["GetArr"] = function(self, cmd, _func)
     local i   = self:c_value(cmd.src_i)
     local dst_typ = cmd.dst_typ
     local line = C.integer(cmd.loc.line)
-    return (util.render([[
-        {
-            pallene_renormalize_array(L, $arr, $i, $line);
-            TValue *slot = &$arr->array[$i - 1];
-            ${check_tag}
-            $dst = $get_slot;
-        } ]], {
-            dst = dst,
-            arr = arr,
-            i = i,
-            line = line,
-            check_tag = self:check_tag(dst_typ, "slot", cmd.loc, "array element"),
-            get_slot = get_slot(dst_typ, "slot"),
-        }))
+
+    local parts = {}
+    table.insert(parts, "{")
+
+    table.insert(parts, util.render([[
+        pallene_renormalize_array(L, $arr, $i, $line);
+        TValue *slot = &$arr->array[$i - 1];]], {
+            arr = arr, i = i, line = line
+    }))
+
+    local check_tag = self:check_tag(dst_typ, "slot", cmd.loc, "array element")
+    if check_tag ~= "" then table.insert(parts, check_tag) end
```

Comment:
I wouldn't bother with this check. Also, I preferred the previous style with a single util.render.

Reply:
Doing it this way has one less `\n` in the end, which makes the generated code look better. (Maybe I should rethink the current strategy of signigicant '\n's)

Reply:
I thought we removed empty lines.

Reply:
I changed that a while ago, now it only merges the empty lines down to a single one.
I'll open an issue to continue the discussion.